SCF-Session-6/intSet/IntSet.java

```diff
@@ -0,0 +1,185 @@
+package intSet;
+
+import java.util.Arrays;
+
+/**
+ * immutable class IntSet which is uses an array to represent a set of integers in the range 1-1000
+ * @author Ishita_Mathur
+ *
+ */
+public final class IntSet 
+{
+	private int[] newArray;
+	final private int arraySize;
+	
+	/**
+	 * Constructor of class
+	 * @param inputArray, an array containing the elements of set
+	 * assume elements are integers
+	 */
+	public IntSet(int inputArray[])
+	{
+		arraySize = removeDuplicate(inputArray);
+		newArray = new int[arraySize];
+		for(int i = 0; i < arraySize; i++)
+			newArray[i] = inputArray[i];
+	}
+	
+	/**
+	 * This method removes the duplicate values from array.
+	 * @param inputArray
+	 * @return size of array after removing duplicate elements
+	 */
+	public int removeDuplicate(int[] inputArray)
+	{
+		int size;
+		
+		Arrays.sort(inputArray);
+		size = inputArray.length;
+		for(int i = 0; i < size - 1; i++)
+		{
```

Comment:
Redesign the method using single for loop. As the array is sorted before removing the duplicate values, it can be resolved using single for loop.

Reply:
done